### DISCUSSION TITLE
Title: Block on Rotating Plate Changes Position but Not Angle

### DISCUSSION THREAD

**Discussion by z-yf17** [google-deepmind/mujoco#1978] [Asking for Help]
Content:

Hi,

I am facing an issue with my MuJoCo simulation and I would appreciate some help.

I have a rotating plate, and when I place a block on it, the block's position changes as the plate rotates, but its angle does not. Ideally, the block should change its angle along with its position as it rotates on the plate.

Why is this happening, and how can I adjust the setup so that the block's angle changes appropriately with the rotation of the plate?

Thank you very much for any assistance!

![Uploading 8fabb63a506d7
<img width="338" alt="65146d4a60c4b24ee38af25614affa5" src="https://github.com/google-deepmind/mujoco/assets/86708745/89aff453-6cf4-4de5-9349-a07c50c67346">
db7b707ba26574d783.png…]()
<img width="386" alt="3a6072c0c624520591ada1b0d5acd45" src="https://github.com/google-deepmind/mujoco/assets/86708745/4e6a0915-8ed4-446d-b769-17e7a2291e4b">
<img width="449" alt="148f3c0cdb778f380a8b592cb7d7e05" src="https://github.com/google-deepmind/mujoco/assets/86708745/642093f1-5c39-44df-9aff-2fd36d9a0e48">

**Comment by Balint-H** [comment]
Can you share a minimum working example XML of only your spinning plate and cube where this effect is reproduced?

**Comment by z-yf17** [comment]
![image](https://github.com/google-deepmind/mujoco/assets/86708745/32aefa22-161a-4990-82e2-04ec9a453c61)
![image](https://github.com/google-deepmind/mujoco/assets/86708745/9aff67b9-4c0c-4bc7-b55c-66268953d289)

Here is the code. Thanks.

**Comment by Balint-H** [comment]
It is caused by the large diagonal inertia you set (by a couple orders of magnitude larger than usual for a cube that size) and the frictionloss you specified in the joint (which resists all movement)

Unless you have a specific reason to use inertial (e.g. measurements from CAD software or real life experiments) I recommend using the default inertia calculations and removing the inertial element.

In the future, please paste MJCFs in a code block, or upload a txt file containing it, or a zip file of the xml. (I know, it is weird that github doesn't support xml files by default)
